src/main/java/com/amazon/ion/impl/bin/IonManagedWriter_1_1.kt

```diff
@@ -339,14 +337,11 @@ internal class IonManagedWriter_1_1(
         val hasMacrosToRetain = macroTable.isNotEmpty()
         if (!hasMacrosToAdd && !hasMacrosToRetain) return
```

Comment:
```suggestion
        if (!hasMacrosToAdd) return
```
When we were writing the symbol and macro tables in the same encoding directive, we had to explicitly retain the existing macros. Using the `set`/`add` macros, we don't have to do that.

Basically, the change I'm suggesting will avoid emitting something like this:
```
(:add_symbols "foo" "bar")
(:add_macros )             // <-- this is just wasted bytes
```

Reply:
You're right that `hasMacrosToRetain` is irrelevant here, and then I can remove the latter check on `hasMacrosToAdd` alone.